commit 0e98738d78b9
Author: Xidorn Quan <me@upsuper.org>
Date:   Mon Jul 17 15:10:50 2017 +1000

    Bug 1338764 part 1 - Use const rather than enum for context flags of nsStyleSVG. r=manishearth
    
    MozReview-Commit-ID: KzORnfVXgLJ
    
    --HG--
    extra : rebase_source : 365842d6e6fda4b35a5932272773a3d88bfc04c6
---
 layout/style/nsStyleStruct.h | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git layout/style/nsStyleStruct.h layout/style/nsStyleStruct.h
index 59833e37f3bb..eccc02380f7e 100644
--- layout/style/nsStyleStruct.h
+++ layout/style/nsStyleStruct.h
@@ -3603,15 +3603,19 @@ private:
   // Flags to represent the use of context-fill and context-stroke
   // for fill-opacity or stroke-opacity, and context-value for stroke-dasharray,
   // stroke-dashoffset and stroke-width.
-  enum {
-    FILL_OPACITY_SOURCE_MASK   = 0x03,  // fill-opacity: context-{fill,stroke}
-    STROKE_OPACITY_SOURCE_MASK = 0x0C,  // stroke-opacity: context-{fill,stroke}
-    STROKE_DASHARRAY_CONTEXT   = 0x10,  // stroke-dasharray: context-value
-    STROKE_DASHOFFSET_CONTEXT  = 0x20,  // stroke-dashoffset: context-value
-    STROKE_WIDTH_CONTEXT       = 0x40,  // stroke-width: context-value
-    FILL_OPACITY_SOURCE_SHIFT   = 0,
-    STROKE_OPACITY_SOURCE_SHIFT = 2,
-  };
+
+  // fill-opacity: context-{fill,stroke}
+  static const uint8_t FILL_OPACITY_SOURCE_MASK   = 0x03;
+  // stroke-opacity: context-{fill,stroke}
+  static const uint8_t STROKE_OPACITY_SOURCE_MASK = 0x0C;
+  // stroke-dasharray: context-value
+  static const uint8_t STROKE_DASHARRAY_CONTEXT   = 0x10;
+  // stroke-dashoffset: context-value
+  static const uint8_t STROKE_DASHOFFSET_CONTEXT  = 0x20;
+  // stroke-width: context-value
+  static const uint8_t STROKE_WIDTH_CONTEXT       = 0x40;
+  static const uint8_t FILL_OPACITY_SOURCE_SHIFT   = 0;
+  static const uint8_t STROKE_OPACITY_SOURCE_SHIFT = 2;
 
   uint8_t          mContextFlags;     // [inherited]
 };

commit 63d6f06939ee
Author: Xidorn Quan <me@upsuper.org>
Date:   Tue Aug 1 16:05:04 2017 +1000

    Bug 1338764 part 2 - Add context-{fill,stroke}-opacity support to {fill,stroke}-opacity. r=manishearth
    
    MozReview-Commit-ID: HzUHmZb8ICl
    
    --HG--
    extra : rebase_source : b8746c4d9d4cfc3abe6645f583dc08d0f7c75341
---
 layout/style/ServoBindings.toml | 2 ++
 1 file changed, 2 insertions(+)

diff --git layout/style/ServoBindings.toml layout/style/ServoBindings.toml
index 5e09c24eb0e3..6d8e979fdc1f 100644
--- layout/style/ServoBindings.toml
+++ layout/style/ServoBindings.toml
@@ -232,6 +232,7 @@ whitelist-types = [
     "nsStylePadding",
     "nsStylePosition",
     "nsStyleSVG",
+    "nsStyleSVGOpacitySource",
     "nsStyleSVGPaint",
     "nsStyleSVGReset",
     "nsStyleTable",
@@ -458,6 +459,7 @@ structs-types = [
     "nsStylePosition",
     "nsStyleQuoteValues",
     "nsStyleSVG",
+    "nsStyleSVGOpacitySource",
     "nsStyleSVGPaint",
     "nsStyleSVGReset",
     "nsStyleTable",

commit 03e036aff26d
Author: Xidorn Quan <me@upsuper.org>
Date:   Tue Aug 1 16:38:27 2017 +1000

    Bug 1338764 part 3 - Update test expectations. r=manishearth
    
    MozReview-Commit-ID: HzN5WP210G7
    
    --HG--
    extra : rebase_source : c3585958cb5159a242515a80f226e43f506319d8
---
 layout/style/test/stylo-failures.md | 8 --------
 1 file changed, 8 deletions(-)

diff --git layout/style/test/stylo-failures.md layout/style/test/stylo-failures.md
index 0509269b1334..d69fa898d9be 100644
--- layout/style/test/stylo-failures.md
+++ layout/style/test/stylo-failures.md
@@ -34,10 +34,6 @@ to mochitest command.
 * Animation support:
   * SMIL Animation
     * test_restyles_in_smil_animation.html [2]
-* Unsupported values
-  * SVG-in-OpenType values not supported servo/servo#15211 bug 1338764
-    * test_value_storage.html `context-` [7]
-    * test_bug798843_pref.html [3]
 * Incorrect serialization
   * place-{content,items,self} shorthands bug 1363971
     * test_align_shorthand_serialization.html [6]
@@ -58,7 +54,3 @@ to mochitest command.
 ## Unknown / Unsure
 
 ## Ignore
-
-* Ignore for now since should be mostly identical to test_value_storage.html
-  * test_value_cloning.html [*]
-  * test_value_computation.html [*]
